### ISSUE TITLE
Support Stagenet on same Monerujo App

### ISSUE THREAD

**Issue by leonardochaia** [m2049r/xmrwallet#528]
Dear Monerujo team:
As far as I can tell, in order to use Monerujo on the Stagenet users would need to install a different apk `monerujo-1x10x14-stage_universal.apk`.
It'd be great if the main application would allow users to select Stagenet somehow, in "advanced options" or something.

I'd love to see this happen for two main reasons:
1. newcomers can start trading with coins without real value immediately.
It'd be great if I could tell this friend who wants to know more about crypto: "download this app, select stagenet and we'll transact some fake coins for you to get used to it."
2. I'm working on a project where people will be able to perform "interactive tutorials" using Monero on Stagenet, so that no real funds are actually spent.

I've attempted to connect to a Stagenet node on Monerujo (from Google Play) and the node has been added correctly, but when I generated a new wallet, the address started with 4 and it didn't actually work.
Perhaps the client should warn you about connecting to a Stagenet node when Stagenet is not selected, or maybe even reject Stagenet nodes when Stagenet is not selected.

I may be able to throw a PR with this, but honestly I haven't coded Java/Android in a couple years and I'm pretty rusty. Although I may be able to achieve it with some help, haven't really looked at the code yet.

Thank you very much, and keep up the good work.
Leo.

**Comment by m2049r** [comment]
i understand all your points. but there are so many problems with stagenet that i think you should be using mainnet for both (1) and (2). especially when you consider that fees are < 0.01 USD. just give the newcomers a dollar worth of monero and be done with it.
even creating a stagenet wallet is problem because the recovery height is wrong, so you **must restore from seed** in any case - this is crap for newcomers!

**Comment by leonardochaia** [comment]
Thank you very much for your answer. Unfortunately using real Moneroj won't do it.
I have tested the oficcial Monero wallet and I could simply create a new wallet for stagenet. I did not have to recover from seed. I tested this using a remote node too.

Is this something particular to Monerujo, and perhaps other wallets?
Could you kindly elaborate more on this last topic?
I understand what you mean by restoring from seed after certain height. Why doesn't this work on stagenet?
Thank you.

**Comment by m2049r** [comment]
> Unfortunately using real Moneroj won't do it.

why not?

restoring from seed works on stagenet. i am just saying a fresh wallet usually does not work and needs to be recovered because the height is wrong. this is because monerujo does not connect to the node in order to create the wallet - and so cannot check the height.

also, since hardly anyone runs stagenet public daemons and even if, not on port 18089 the "node-o-matic" node management does not work in stagenet.

also, some bitcoin features like bitpay don't work in stagenet because bitpay does not provide a test infrastructure.

**TL;DR: Stagenet implementation in monerujo is not targeting the enduser but for development testing. The endusers should use mainnet with real moneroj.**

**Comment by bitOMG** [comment]
Stagenet the name could also be called testnet ? That will be more comprehensive by newby.

**Comment by m2049r** [comment]
stagenet is not testnet.
